source/slang/slang-ir-glsl-legalize.cpp

```diff
@@ -3936,4 +3936,41 @@ void legalizeDynamicResourcesForGLSL(CodeGenContext* context, IRModule* module)
     }
 }
 
+void legalizeFunctionForEntryPointAttributesForGLSL(IRModule* module, IRFunc* entryPointFunc)
+{
+    // Dummy builder required to create decorations.
+    IRBuilder builder(module);
+    builder.setInsertInto(entryPointFunc);
+
+    for (auto globalInst : module->getGlobalInsts())
```

Comment:
This should be traversing the call graph of each EntryPoint, otherwise we will be marking all entrypoints with the decorations if any one function in the module has the opcode.

Reply:
See buildEntryPointReferenceGraph, it would be best if we can perform this logic in a place where the reference graph is already available.